### ISSUE TITLE
Dawntrail Update Checklist / Tracking

### ISSUE THREAD

**Issue by wexxlee** [OverlayPlugin/cactbot#170]
This issue serves as general checklist of necessary updates and (anticipated) content we intend to support following the release of Patch 7.0.  The checklist will be updated as needed with additional work items/content changes after patch notes are released or as additional work items are identified.

Please refer to the [Patch Update Checklist](https://github.com/OverlayPlugin/cactbot/blob/main/docs/PatchUpdateChecklist.md) for more info.

> [!IMPORTANT]
> **If you intend to work on something below, please edit this issue to add yourself next to the item, or reply below so we can add you.  This helps avoid unnecessary duplication of effort.**
>
> PRs addressing items below should reference this issue for tracking purposes.

(Please feel free to edit this post to add additional updates/content that may be missing!)

### Downtime

- [ ] Add `07-dt` to config UI
- [ ] Add `exVersion=5` to coverage
- [ ] Add VPR/PCT to job enums & types

### Post-Game Download
- [ ] Add VPR/PCT job icons to `resources`
- [ ] Update game data resources files (pending XIVAPI)

### Post-Game Live
- [ ] Check/update memory sigs & offsets (coordinate with OverlayPlugin for charmap & in combat sigs)
- [ ] Update `content_list.ts` with new (ordered) content
- [ ] Update oopsy's `buff_map.ts` with job buff changes
- [ ] Add/update buff icons and update `buff_tracker.ts` in Jobs module
- [ ] Add components for VPR/PCT to Jobs module

----

### Patch 7.0 Content Support (raidboss/oopsy)
> [!NOTE]
> A separate tracking issue will be created for Patch 7.01 & 7.05 content.

#### Dungeons
- [ ] msq1 (lv. 91) dungeon
- [ ] msq2 (lv. 93) dungeon
- [ ] msq3 (lv. 95) dungeon
- [ ] msq4 (lv. 97) dungeon
- [ ] msq5 (lv. 99) dungeon
- [ ] msq6 (lv. 100) dungeon
- [ ] optional1 (lv. 100) dungeon
- [ ] optional2 (lv. 100) dungeon

> Note: Probably need a non-spoilery way to distinguish these last two for coordination purposes.

#### Trials
- [ ] trial1 (lv. 93) normal trial
- [ ] trial2 (lv. 99) normal trial
- [ ] trial3 (lv. 100) normal trial
- [ ] ex1 (lv. 100) extreme trial
- [ ] ex2 (lv. 100) extreme trial
- [ ] ex3 (lv. 100) extreme trial

#### Misc.
- [ ] hunts

**Comment by valarnin** [comment]
Re: XIVAPI - There's a new beta for XIVAPI at https://beta.xivapi.com. I'm not sure which API will get updates first, the way the Discord announcement reads makes it sounds like the beta might get updates first as it's easier to update.

`ex3` is usually not available at release, I believe, based on previous patches.

**Comment by wexxlee** [comment]
> Re: XIVAPI - There's a new beta for XIVAPI at https://beta.xivapi.com. I'm not sure which API will get updates first, the way the Discord announcement reads makes it sounds like the beta might get updates first as it's easier to update.

Good timing on that.  I'll poke around this weekend and get a PR ready to swap us over to the beta server just in case.

> `ex3` is usually not available at release, I believe, based on previous patches.

Oh, right, I completely forgot about the delay on ex3.  I'll yank it from the list (can always re-add if they do something different here.)

**Comment by g0shu** [comment]
Ablilty ids for oopsy's overwritten mit tracker will need to be updated too. 
https://github.com/OverlayPlugin/cactbot/blob/9899675e9d26c721fe1af07901864fc92ef81283/ui/oopsyraidsy/data/00-misc/general.ts#L35-L74
I'll get around to it eventually if nobody else does, but probably not until a few days before savage release.

**Comment by wexxlee** [comment]
> Ablilty ids for oopsy's overwritten mit tracker will need to be updated too.
 
Added!

**Comment by xiashtra** [comment]
sounds like we'll be getting changes to a few log lines per [Ravahn's beta post](https://discord.com/channels/551474815727304704/551476805148737546/1256332093697429615):
> - Modified 21/22 loglines to include the combat action source's pet owner id / name.
> - Modified 21/22 loglines to include the orientation field (in hex, for now).  My hope is that this will assist with positional damage calculation in the future
> - Modified 21/22 loglines to include separate effect display type, animation id, animnation delay, and action id fields
> - Modified 12 loglines (player stats) to remove an ancient hard-coded '0' just before tenacity & local content id.

**Comment by MaikoTan** [comment]
Need a checklist for every jobs in jobs module too. I think it can be listed here rather than open a separate issue.

## Common

- [x] Version Diff Process ( #204 )
- [ ] Buff Tracker
- [ ] Combo Tracker
- [ ] Player Speed Buffs

## Job-specific Component

### Tanks

- [ ] PLD
- [ ] WAR
- [ ] DRK
- [ ] GNB

### Healers

- [ ] WHM
- [ ] SCH
- [ ] AST
- [ ] SGE

### Melee

- [ ] MNK
- [ ] DRG
- [ ] NIN
- [ ] SAM
- [ ] RPR
- [ ] VPR (new!)

### Ranged

- [ ] BRD
- [ ] MCH
- [ ] DNC

### Magic

- [ ] BLM
- [ ] SMN
- [ ] RDM
- [ ] PCT (new!)
- [ ] ~~BLU~~ (?)

**Comment by JLGarber** [comment]
I'll put my name down for MSQ 100, since there's nobody else visibly on it. I'll likely start with basic basic trigger and timeline support and flesh it out over this next week.

**Comment by wexxlee** [comment]
> UPDATE: After two weeks, we finally completed all job components in the jobs module, thanks to hard-working @Echoring and no-so-hard-working @MaikoTan (`⁄(⁄⁄-⁄ω⁄-⁄⁄)⁄ﾃﾚﾃﾚ`)! Since we have no every jobs in level 100, some of them are even foreign to us, we cannot 100 percent confirm that those modules are perfectly implemented. If you encounter any problem, please open an issue and let me or @Echoring know, merci beaucoup !

Thanks so much to both of you @MaikoTan and @Echoring for all the work on this! ❤️

**Comment by valarnin** [comment]
(Might want to update the tracking post to indicate that raid content doesn't have oopsy support yet)

**Comment by wexxlee** [comment]
> (Might want to update the tracking post to indicate that raid content doesn't have oopsy support yet)

Done!  (And feel free to edit directly if you spot anything else -- any collaborator is able to edit the original post!)

**Comment by wexxlee** [comment]
Going to unpin this, since I don't know if anyone will get around to doing LHWS oopsy triggers or Tender Valley.  (I was holding on doing oopsy until I had time to clean up and merge #475, but I haven't gotten back to it yet.)

Will leave the issue open though.  Maybe someday.
